lib/Fenix/Model/IRC.pm

```diff
@@ -0,0 +1,121 @@
+package Fenix::Model::IRC;
+use Mojo::Base -base, -signatures;
+
+use Time::Piece;
+
+use Mojo::Exception;
+use Mojo::File;
+use Mojo::IRC;
+use Mojo::JSON qw(encode_json);
+
+# attributes
+has config  => sub { {} };
+has datadir => sub { Mojo::Exception->throw("ERROR: datadir must be specified on instantiation.\n") };
+has chans   => sub {
+    return {
+        map {
+            $_->{name} => {
+                log         => $_->{log}         eq 'on' ? 1 : 0,
+                interactive => $_->{interactive} eq 'on' ? 1 : 0,
+            }
+        } @{shift->config->{CHANS}}
+    }
+};
+
+has irc => sub($self) {
+    Mojo::IRC->new(
```

Comment:
have you considered inheriting from Mojo::IRC instead instead of adding an instance?